src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -327,58 +327,8 @@ const useGroupInfo = () => {
   };
 
   const handleMessageAnonymously = async () => {
-    if (!selectedUser?.allow_anon_dm) {
-      SimpleToast.show('This user does not allow anonymous messages');
-      return;
-    }
-
-    try {
-      setOpenModal(false);
-      const response = await getOrCreateAnonymousChannel(selectedUser?.user_id);
-      const targetRawJson = {
-        type: 'notification.message_new',
-        cid: response?.channel?.id,
-        channel_id: '',
-        channel_type: 'messaging',
-        channel: response?.channel,
-        created_at: response?.channel,
-        targetName: selectedUser?.user?.name,
-        targetImage: selectedUser?.user?.image
-      };
-      const channelList = ChannelList.fromMessageAnonymouslyAPI({
-        channel: response?.channel,
-        members: response?.members,
-        appAdditionalData: {
-          rawJson: targetRawJson,
-          message: '',
-          targetName: selectedUser?.user?.name,
-          targetImage: selectedUser?.user?.image
-        }
-      });
-
-      await channelList.saveIfLatest(localDb);
-      try {
-        response?.members?.forEach(async (member) => {
-          const userMember = UserSchema.fromMemberWebsocketObject(member, response?.channel?.id);
-          await userMember.saveOrUpdateIfExists(localDb);
-
-          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
-            response?.channel?.id,
-            uuid(),
-            member
-          );
-          await memberSchema.save(localDb);
-        });
-      } catch (e) {
-        console.log('error on memberSchema');
-        console.log(e);
-      }
-
-      setOpenModal(false);
-      goToChatScreen(channelList);
-    } catch (e) {
-      SimpleToast.show(e || 'Failed to message this user anonymously');
-    }
+    setOpenModal(false);
+    handleAnonymousMessage(selectedUser);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for handling anonymous messages has been simplified. The new function `handleAnonymousMessage` is being used instead of the previous complex logic. However, please ensure that the new function handles all edge cases and performs all necessary operations.

```diff
- const response = await getOrCreateAnonymousChannel(selectedUser?.user_id);
- // ... rest of the code
+ handleAnonymousMessage(selectedUser);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->